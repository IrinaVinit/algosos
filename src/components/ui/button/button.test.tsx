import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from "./button";
import renderer from "react-test-renderer";

const buttonText ="Click me";

describe("test Button component", ()=>{
    it("render with text", ()=>{
        const tree = renderer.create(<Button text={buttonText}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render without text", ()=>{
        const tree = renderer.create(<Button/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("render desabled", ()=>{
        const tree = renderer.create(<Button disabled/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it("loading", ()=>{
        const tree = renderer.create(<Button isLoader/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
    
  it("call callback", () => {
    const click = jest.fn();
    render(<Button onClick={click} text={buttonText}/>);
    const button = screen.getByText(buttonText);
    fireEvent.click(button);
    expect(click).toHaveBeenCalledTimes(1);
  });
});


