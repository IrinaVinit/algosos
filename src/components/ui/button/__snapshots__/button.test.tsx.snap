// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`test Button component render desabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`test Button component render with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Click me
  </p>
</button>
`;

exports[`test Button component render without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
